packages/api/AlvTime.Persistence/Repositories/AccessTokenStorage.cs

```diff
@@ -1,70 +1,63 @@
-﻿using AlvTime.Business.AccessToken;
-using AlvTime.Persistence.DataBaseModels;
+﻿using AlvTime.Persistence.DataBaseModels;
 using System;
 using System.Collections.Generic;
-using System.Globalization;
 using System.Linq;
+using AlvTime.Business.AccessTokens;
+using AlvTime.Business.Interfaces;
+using AlvTime.Business.Models;
 
 namespace AlvTime.Persistence.Repositories
 {
     public class AccessTokenStorage : IAccessTokenStorage
     {
-        private readonly AlvTime_dbContext _context;
+        private readonly AlvTime_dbContext _dbContext;
 
-        public AccessTokenStorage(AlvTime_dbContext context)
+        public AccessTokenStorage(AlvTime_dbContext dbContext)
         {
-            _context = context;
+            _dbContext = dbContext;
         }
 
-        public AccessTokenResponseDto CreateLifetimeToken(string friendlyName, int userId)
+        public AccessTokenDto CreateLifetimeToken(string friendlyName, GenericUser user)
         {
-            var uuid = Guid.NewGuid().ToString();
-
+            var alvUser = _dbContext.User.First(dbUser => dbUser.Email.ToLower().Equals(user.Email.ToLower()));
+            
             var accessToken = new AccessTokens
             {
-                UserId = userId,
-                Value = uuid,
+                UserId = alvUser.Id,
```

Comment:
Nice! I have a few suggestions here: Since we're trying out the "onion" architecture, I guess most of the business logic should be placed inside the business project. Now, what is defining the business logic? Well, that is the hard part. I would argue that the action to create a value and expiry date should be considered as business logic for lifetime token creation. In my opinion, I think I would have made this repository code something like this:

```
public Task CreateLifetimeToken(LifetimeToken token)
{
    await _dbContext.AccessTokens.AddAsync(new AccessTokenEntity
    {
        UserId = token.User.Id,
        Value = token.Value,
        ExpiryDate = token.ExpiryDate,
        FriendlyName = token.FriendlyName
    });

    await _dbContext.SaveChangesAsync();
}
```

Then leave the responsibility to actually create such lifetime token to the business layer